main.tf

```
@@ -111,3 +111,12 @@ resource "aws_route_table_association" "private" {
   subnet_id      = aws_subnet.private[count.index].id
   route_table_id = aws_route_table.private[count.index].id
 }
+
+terraform {
+  backend "s3" {
+    bucket         = "terraform-mentordevops-state"
+    key            = "home/techstellar/DevOps/Aws-terra-docker/firstkeyaws.pem"
```

This is not your pem key. It's a key in s3 where tf file will be stored
```suggestion
    key            = "main.tfstate"
```